spec/Section 6 -- Execution.md

```diff
@@ -477,8 +487,9 @@ The depth-first-search order of the field groups produced by {CollectFields()}
 is maintained through execution, ensuring that fields appear in the executed
 response in a stable and predictable order.
 
-CollectFields(objectType, selectionSet, variableValues, visitedFragments):
+CollectFields(objectType, selectionSet, variableValues, taggedMemberFieldName, visitedFragments):
 
+  * If {taggedMemberFieldName} is not provided, initialize it to {null}.
```

Comment:
Clever - but what might simplify is having a step at the end along the lines of "If this type is tagged, then groupedFields must include exactly one entry"

Reply:
Oh wait, maybe I'm misunderstanding this... you want to query as many or few fields as desired in a tagged output? It's just that the executed result map only includes one key for that particular selection set?

Reply:
This makes it explicit that `taggedMemberFieldName` is optional; it's not needed when asserting there's only one field on the subscription root (for example) which still just calls `CollectFields(subscriptionType, selectionSet, variableValues)` as it did before.

We need the null default so we can use the logic `If {taggedMemberFieldName} is not null` below to opt into the tagged behaviour. (In `ExecuteSelectionSet`, `taggedMemberFieldName` will be non-null iff the selection set is on a tagged type.)

> you want to query as many or few fields as desired in a tagged output?

Yes

> It's just that the executed result map only includes one key for that particular selection set?

I'm not certain enough on the terminology `executed result map` to confirm this. The following is valid fragment against a tagged type `tagged output MyTaggedType { cat: Cat, dog: Dog, fish: Fish }` (note aliases are fine):

```graphql
fragment Foo on MyTaggedType {
  __typename
  cat1: cat { name }
  cat2: cat { name id }
  cat3: cat { id numberOfLives }
  dog { id name waggingTail }
}
```

When we `ExecuteSelectionSet` on a tagged type, we use `GetTaggedMemberFieldName` to determine which key was actually set from the resolver (i.e. the object returned from the resolver should be  `{cat: {...}}` or `{dog: {...}}` or `{fish: {...}}`); this key (`cat`, `dog` or `fish`) is the `taggedMemberFieldName` and we then automatically skip over all selections that don't relate to this field (similar to `@skip(if: true)`). So in the case of `{cat: {...}}` we'd get `taggedMemberFieldName = 'cat'`, and the result of the fragment would be:

```json
{
  "__typename": "MyTaggedType",
  "cat1": { "name": "Brontie" },
  "cat2": { "name": "Brontie", "id": 2 },
  "cat3": { "id": 2, "numberOfLives": 8 }
}
```